staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -4996,11 +4996,21 @@ type EndpointPort struct {
 	// +optional
 	Protocol Protocol `json:"protocol,omitempty" protobuf:"bytes,3,opt,name=protocol,casttype=Protocol"`
 
-	// The application protocol for this port.
+	// Used as a hint for implementations to
+	// configure the protocol used between the
+	// implementation and the application it exposes.
 	// This field follows standard Kubernetes label syntax.
-	// Un-prefixed names are reserved for IANA standard service names (as per
+	// Valid values are either:
+	//
+	// * Un-prefixed protocol names - reserved for IANA standard service names (as per
 	// RFC-6335 and https://www.iana.org/assignments/service-names).
-	// Non-standard protocols should use prefixed names such as
+	//
+	// * Kubernetes standard names:
+	//   * 'k8s.io/h2c' - http2 over cleartext as described in https://www.rfc-editor.org/rfc/rfc7540
```

Comment:
"HTTP/2" ?

Also - do we need HTTP/2 over TLS ?

Reply:
"HTTP/2" implies TLS by default, see section 3 of the rfc
```
[3.1](https://www.rfc-editor.org/rfc/rfc7540#section-3.1).  HTTP/2 Version Identification

   The protocol defined in this document has two identifiers.

   o  The string "h2" identifies the protocol where HTTP/2 uses
      Transport Layer Security (TLS) [[TLS12](https://www.rfc-editor.org/rfc/rfc7540#ref-TLS12)].  This identifier is used
      in the TLS application-layer protocol negotiation (ALPN) extension
      [[TLS-ALPN](https://www.rfc-editor.org/rfc/rfc7540#ref-TLS-ALPN)] field and in any place where HTTP/2 over TLS is
      identified.

      The "h2" string is serialized into an ALPN protocol identifier as
      the two-octet sequence: 0x68, 0x32.

   o  The string "h2c" identifies the protocol where HTTP/2 is run over
      cleartext TCP.  This identifier is used in the HTTP/1.1 Upgrade
      header field and in any place where HTTP/2 over TCP is identified.

      The "h2c" string is reserved from the ALPN identifier space but
      describes a protocol that does not use TLS.

   Negotiating "h2" or "h2c" implies the use of the transport, security,
   framing, and message semantics described in this document.
```

Reply:
Sorry, I meant "HTTP/2" in the explanation instead of "http2".

Other part stadns - If we are defining k8s "h2c", should we also define "h2" ?

Reply:
My thinking was that for "over TLS" we should encourage implementations to use `https` follow the ALPN negotiation. Thats what also reflected in the KEP. 
I do not think it is smart to have such thing "TLS only with h2", it is highly depended on the client and the server.
But I am not an expert in this field, maybe @howardjohn or @robscott  would be able to elaborate ?

Reply:
Fix the strings.  We can always add new definitions if needed, seeing as it is literally a comment :)

Code-freeze....

Reply:
I think @LiorLieberman's right here. "h2" would be accessed via ALPN, I think it would be difficult/impossible for many implementations to _only_ support "TLS only with h2".